starlite/middleware/session/sqlalchemy_backend.py

```diff
@@ -0,0 +1,266 @@
+        if session_obj:
+            if not session_obj.expired:
+                self._update_session_expiry(session_obj)  # type: ignore[unreachable]
+                await sa_session.commit()
+                return session_obj.data
+            await sa_session.delete(session_obj)
+            await sa_session.commit()
+        return None
+
+    async def set(self, session_id: str, data: bytes) -> None:
+        """Store `data` under the `session_id` for later retrieval.
+
+        If there is already data associated with `session_id`, replace
+        it with `data` and reset its expiry time
+        """
+        sa_session = self._create_sa_session()
+        session_obj = await self._get_session_obj(sa_session=sa_session, session_id=session_id)
+
+        if not session_obj:
+            session_obj = self._model(session_id=session_id)  # type: ignore[call-arg]
+            sa_session.add(session_obj)
+        session_obj.data = data
+        self._update_session_expiry(session_obj)
+        await sa_session.commit()
+
+    async def delete(self, session_id: str) -> None:
+        """Delete the data associated with `session_id`"""
+        sa_session = self._create_sa_session()
+        await sa_session.execute(sa.delete(self._model).where(self._model.session_id == session_id))
+        await sa_session.commit()
+
+    async def delete_all(self) -> None:
+        """Delete all data stored within this backend."""
+        sa_session = self._create_sa_session()
+
+        await sa_session.execute(sa.delete(self._model))
+        await sa_session.commit()
+
+    async def delete_expired(self) -> None:
+        """Delete all expired session from the database."""
+        sa_session = self._create_sa_session()
+        await sa_session.execute(sa.delete(self._model).where(self._model.expired))
+
+
+class SQLAlchemyBackend(BaseSQLAlchemyBackend[SASession]):
+    def _get_session_obj(self, *, sa_session: SASession, session_id: str) -> Optional[SessionModelMixin]:
+        return sa_session.scalars(self._select_session_obj(session_id)).one_or_none()
+
+    def _get(self, session_id: str) -> Optional[bytes]:
+        sa_session = self._create_sa_session()
+        session_obj = self._get_session_obj(sa_session=sa_session, session_id=session_id)
+
+        if session_obj:
+            if not session_obj.expired:
+                self._update_session_expiry(session_obj)  # type: ignore[unreachable]
+                sa_session.commit()
+                return session_obj.data
+            sa_session.delete(session_obj)
+            sa_session.commit()
+        return None
+
+    async def get(self, session_id: str) -> Optional[bytes]:
+        """Retrieve data associate with `session_id`.
+
+        If no data for the given `session_id` exists, return an empty
+        dict
+        """
+        return await anyio.to_thread.run_sync(self._get, session_id)
+
+    def _set(self, session_id: str, data: bytes) -> None:
```

Comment:
same comment applies here and in the other methods, i.e. `_set_sync`, `_delete_sync` etc.